`react-dom/server could not be resolved` in `react-apollo` in Next@5

```
ERROR in ./node_modules/.registry.npmjs.org/react-apollo/2.1.0-beta.0/node_modules/react-apollo/renderToStringWithData.js
Module not found: Error: Can't resolve 'react-dom/server' in '/Users/Vaughan/dev-live/vwp/node_modules/.registry.npmjs.org/react-apollo/2.1.0-beta.0/node_modules/react-apollo'
 @ ./node_modules/.registry.npmjs.org/react-apollo/2.1.0-beta.0/node_modules/react-apollo/renderToStringWithData.js 3:15-42
 @ ./node_modules/.registry.npmjs.org/react-apollo/2.1.0-beta.0/node_modules/react-apollo/index.js
 @ ./packages/next-apollo/withData.js
 @ ./pages/map.js
 @ multi ./pages/map.js
```

See https://github.com/zeit/next.js/blob/f9b52cfcb6cba5f2af93caf3cc8bc3636571323f/server/build/webpack.js#L155

This aliasing seems to prevent require sub-paths.

PS. I am using `pnpm`. *Maybe* this could be related...

**Workaround**

Add this to your `next.config.js`:

```
delete config.resolve.alias['react-dom']
```

<!--- Provide a general summary of the issue in the Title above -->

<!--
    Thank you very much for contributing to Next.js by creating an issue! ❤️
    To avoid duplicate issues we ask you to check off the following list
-->

<!-- Checked checkbox should look like this: [x] -->
- [x] I have searched the [issues](https://github.com/zeit/next.js/issues) of this repository and believe that this is not a duplicate.


## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
| Tech    | Version |
|---------|---------|
| next    |       5  |
| node    |   8      |
| OS      |     macOS    |
| browser |     Chrome    |
| etc     |         |

--- a/server/build/webpack.js
+++ b/server/build/webpack.js
@@ -149,10 +149,15 @@ export default async function getBaseWebpackConfig (dir, {dev = false, isServer 
       ],
       alias: {
         next: nextDir,
-        // This bypasses React's check for production mode. Since we know it is in production this way.
-        // This allows us to exclude React from being uglified. Saving multiple seconds per build.
-        react: dev ? 'react/cjs/react.development.js' : 'react/cjs/react.production.min.js',
-        'react-dom': dev ? 'react-dom/cjs/react-dom.development.js' : 'react-dom/cjs/react-dom.production.min.js'
+        // React already does something similar to this.
+        // But if the user has react-devtools, it'll throw an error showing that
+        // we haven't done dead code elimination (via uglifyjs).
+        // We purposly do not uglify React code to save the build time.
+        // (But it didn't increase the overall build size)
+        // Here we are doing an exact match with '$'
+        // So, you can still require nested modules like `react-dom/server`
+        react$: dev ? 'react/cjs/react.development.js' : 'react/cjs/react.production.min.js',
+        'react-dom$': dev ? 'react-dom/cjs/react-dom.development.js' : 'react-dom/cjs/react-dom.production.min.js'
       }
     },
     resolveLoader: {

--- a/server/build/webpack.js
+++ b/server/build/webpack.js
@@ -148,11 +148,7 @@ export default async function getBaseWebpackConfig (dir, {dev = false, isServer 
         ...nodePathList // Support for NODE_PATH environment variable
       ],
       alias: {
-        next: nextDir,
-        // This bypasses React's check for production mode. Since we know it is in production this way.
-        // This allows us to exclude React from being uglified. Saving multiple seconds per build.
-        react: dev ? 'react/cjs/react.development.js' : 'react/cjs/react.production.min.js',
-        'react-dom': dev ? 'react-dom/cjs/react-dom.development.js' : 'react-dom/cjs/react-dom.production.min.js'
+        next: nextDir
       }
     },
     resolveLoader: {

--- a/server/build/webpack.js
+++ b/server/build/webpack.js
@@ -148,7 +148,11 @@ export default async function getBaseWebpackConfig (dir, {dev = false, isServer 
         ...nodePathList // Support for NODE_PATH environment variable
       ],
       alias: {
-        next: nextDir
+        next: nextDir,
+        // This bypasses React's check for production mode. Since we know it is in production this way.
+        // This allows us to exclude React from being uglified. Saving multiple seconds per build.
+        react: dev ? 'react/cjs/react.development.js' : 'react/cjs/react.production.min.js',
+        'react-dom': dev ? 'react-dom/cjs/react-dom.development.js' : 'react-dom/cjs/react-dom.production.min.js'
       }
     },
     resolveLoader: {

